From 3d7f30cff855f71ebc7c865d9db31b0873bc7fa2 Mon Sep 17 00:00:00 2001
From: wouterlucas <wouter@wouterlucas.com>
Date: Fri, 5 Jul 2019 15:37:26 -0700
Subject: [PATCH] [CMAKE] Use PYTHON_EXECUTABLE provided by build system

This change is required to make use of OE's python-native package (over the host python) in the nativesdk sysroot and include the jsonref package.
Instead of just calling python and hoping someone answers.

Signed-off-by: wouterlucas <wouter@wouterlucas.com>
---
 CMakeLists.txt                           | 2 +-
 cmake/modules/CmakeHelperFunctions.cmake | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4435eba..589c9f7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ option(BUILD_TESTS "Build tests (requires gtest)" OFF)
 
 set(NAMESPACE ${PROJECT_NAME} CACHE STRING "Namespace of the project")
 
-# TODO: install the generators to staging, for now just use absolute path to this project
+find_package(PythonInterp 2.7 REQUIRED)
 set(JSON_GENERATOR "${PROJECT_SOURCE_DIR}/Tools/JsonGenerator/JsonGenerator.py")
 set(PROXYSTUB_GENERATOR "${PROJECT_SOURCE_DIR}/Tools/ProxyStubGenerator/StubGenerator.py")
 
diff --git a/cmake/modules/CmakeHelperFunctions.cmake b/cmake/modules/CmakeHelperFunctions.cmake
index 312d429..b0c904e 100644
--- a/cmake/modules/CmakeHelperFunctions.cmake
+++ b/cmake/modules/CmakeHelperFunctions.cmake
@@ -714,7 +714,7 @@ function(JsonGenerator)
     endif()
 
     foreach(_input ${Argument_INPUT})
-        execute_process(COMMAND ${_execute_command} ${_input} RESULT_VARIABLE rv)
+        execute_process(COMMAND ${PYTHON_EXECUTABLE} ${_execute_command} ${_input} RESULT_VARIABLE rv)
         if(NOT ${rv} EQUAL 0)
             message(FATAL_ERROR "JsonGenerator generator failed.")
         endif()
@@ -781,7 +781,7 @@ function(ProxyStubGenerator)
     endif()
 
     foreach(_input ${Argument_INPUT})
-        execute_process(COMMAND ${_execute_command} ${_input} RESULT_VARIABLE rv)
+        execute_process(COMMAND ${PYTHON_EXECUTABLE} ${_execute_command} ${_input} RESULT_VARIABLE rv)
         if(NOT ${rv} EQUAL 0)
             message(FATAL_ERROR "ProxyStubGenerator generator failed.")
         endif()
-- 
2.11.0

